recipe/meta.yaml

```diff
@@ -11,50 +15,80 @@ source:
 
 build:
   noarch: python
-  script: {{ PYTHON }} -m pip install . -vv
-  number: 0
-
-requirements:
-  host:
-    - python >=3.9
-    - setuptools >=42
-    - setuptools-scm >=7
-    - pip
-  run:
-    - python >=3.9
-    - numpy >=1.23
-    - pandas >=2.0
-    - packaging >=23.1
-  run_constrained:
-    - bottleneck >=1.3
-    - cartopy >=0.21
-    - cftime >=1.6
-    - dask-core >=2023.4
-    - distributed >=2023.4
-    - flox >=0.7
-    - h5netcdf >=1.1
-    - h5py >=3.8
-    - hdf5 >=1.12
-    - iris >=3.4
-    - matplotlib-base >=3.7
-    - nc-time-axis >=1.4
-    - netcdf4 >=1.6.0
-    - numba >=0.56
-    - pint >=0.22
-    - scipy >=1.10
-    - seaborn >=0.12
-    - sparse >=0.14
-    - toolz >=0.12
-    - zarr >=2.14
+  number: 1
 
-test:
-  imports:
-    - xarray
-    - xarray.backends
-  requires:
-    - pip
-  commands:
-    - pip check
+outputs:
+  - name: xarray-core
```

Comment:
A while back we went from a fully loaded xarray to a "core" but the upstream devs decided for a breaking change and keep the xarray name instead of creating xarray-core. In a way, this is another breaking change. While the patter here is common in conda-forge\*, I would go for xarray to be the same and avoid another breaking change and add a `xarray-extras` (or better yet, use the [optional names upstream](https://github.com/pydata/xarray/blob/f15082c90180a38e485d7362162f05a7e4a6b147/pyproject.toml#L31-L48).)

\* The main reason for that is b/c we, conda-forge, tried to ship packages with as much extras we could but we did not had the multiple outputs at the time. When multiple outputs landed, we used the `-core` pattern to avoid breaking changes. Xarray was an exception at the request of the upstream devs.

Reply:
see pydata/xarray#9149 for the reasoning of the change

Reply:
So the issue is that current `xarray` is just the minimal 

```
    - python >=3.9
    - numpy >=1.23
    - pandas >=2.0
    - packaging >=23.1
```

while the rest of the packages is just introduced via `run_constrained`.

I see. So the impact for current users is, that they would get the new `xarray-core` plus the new 

```
        # accel
        - flox >=0.7
        - opt_einsum
        - numbagg
        # io & cloud
        - fsspec
        - netcdf4 >=1.6.0
        - zarr >=2.14
        # plotting
        - matplotlib-base >=3.7
        # misc
        - pooch
```

Not sure what to do now.

Reply:
I wish they were at this position a few years back, when I did not want to make a breaking change. So, here we go, another breaking change! Go for it. At this point I lost the will to try to communicate what the implications of these changes are to the users and the maintainers here and upstream need to figure this out by themselves.

Reply:
@ocefpaf So along what gdal is doing atm? 

As far as I'm concerned I would be good with

- xarray ( `xarray-core` in this PR)
- xarray-minimal ( `xarray` in this PR)
- xarray-accel
- xarray-io
- xarray-parallel
- xarray-viz
- xarray-complete

So xarray core package would stay as is -> xarray. No breaking changes. :-)

Reply:
@dcherian, @shoyer, @max-sixty, do you have any thoughts here?

Reply:
> As far as I'm concerned I would be good with

LGTM. I would not create the name `-minimal` to avoid confusion but as long as the `xarray` one doesn't change and the others reflect what is in the pyproject.toml, I'm OK. That not only avoid another breaking change but also mimics the original intention in the PyPI package.

Reply:
Thanks for your help here @ocefpaf. Our thinking has moved quite a bit in response to the many support issues we've been seeing.

----

FWIW my original proposal was to have a 3-6 month deprecation cycle where both `xarray` and `xarray-core` meant the same thing to give enough time for the users that care about minimal dependencies to move away.

Reply:
To the extent that some burden falls on conda, I can see that our choices are annoying. It sounds like this isn't the first our of changes, which should bring us pause.

That said, even taking into account that pause, I'm fairly confident that the default experience when using `xarray` should be a package with enough dependencies for it to operate well, and that means including the "highly recommended dependencies".

I don't see how we can realistically make something like `xarray-plus` the default experience, so I don't see any other alternative than making `xarray` have the "highly recommended dependencies". Do others?

(even if I'm correct, it doesn't mean that we did this well, and so apologies to @ocefpaf...)

Reply:
Let's ignore what it was but focus on what it can be. My only question is: why not the same as the package published on PyPI to ensure conda-forge and pip users will have a similar experience?

Reply:
I very much thought this would be the same change for PyPI and conda.

(Unless someone thought differently? Strong agree with @ocefpaf that they should be the same...)